cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx

```diff
@@ -0,0 +1,147 @@
+import { ITableData } from 'components/PreviewData';
+import RecordNavigator from 'components/PreviewData/RecordView/Navigator';
+import RecordTable from 'components/PreviewData/RecordView/RecordTable';
+import { INode } from 'components/PreviewData/utilities';
+import If from 'components/If';
+import { styles as tableStyles } from 'components/PreviewData/DataView/TableContainer';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import T from 'i18n-react';
+import classnames from 'classnames';
+
+const I18N_PREFIX = 'features.PreviewData.RecordView.RecordContainer';
+
+const styles = (theme): StyleRules => ({
+  ...tableStyles(theme),
+});
+
+interface IRecordViewContainerProps extends WithStyles<typeof styles> {
+  tableData: ITableData;
+  selectedNode: INode;
+  previewStatus?: string;
+}
+
+const RecordViewBase: React.FC<IRecordViewContainerProps> = ({
+  classes,
+  tableData,
+  selectedNode,
+}) => {
+  const [selectedRecord, setRecord] = useState(1);
+  const [activeTab, setActiveTab] = useState(null);
+
+  const inputs = tableData.inputs;
+  const outputs = tableData.outputs;
+
+  const numRecords = Math.max(tableData.inputFieldCount, tableData.outputFieldCount);
+
+  const updateRecord = (newVal: string) => {
+    const recordNum = parseInt(newVal.split(' ')[1], 10);
+    setRecord(recordNum);
+  };
+
+  const showInputTabs = inputs.length > 1;
+  const showOutputTabs = outputs.length > 1;
+
+  const handleTabClick = (id) => {
+    setActiveTab(id);
+  };
+
+  const getTabConfig = (stagesInfo, recordNum: number) => {
+    const recordIndex = recordNum - 1;
+    const tabs = stagesInfo.map(([stageName, recordInfo], index) => {
+      return {
+        id: index + 1,
+        name: stageName,
+        content: (
+          <RecordTable headers={recordInfo.schemaFields} record={recordInfo.records[recordIndex]} />
+        ),
+        paneClassName: 'record-pane',
+      };
+    });
+    return {
+      defaultTab: 1,
+      layout: 'horizontal',
+      tabs,
+    };
+  };
+
+  const getTabs = (config) => {
+    return <ConfigurableTab tabConfig={config} onTabClick={handleTabClick} activeTab={activeTab} />;
+  };
+
+  return (
+    <div>
+      <RecordNavigator
+        selectedRecord={selectedRecord}
+        numRecords={numRecords}
+        updateRecord={updateRecord}
+        prevOperation={() => setRecord(selectedRecord - 1)}
+        nextOperation={() => setRecord(selectedRecord + 1)}
+      />
+      <div className={classes.outerContainer}>
+        <If condition={!selectedNode.isSource && !selectedNode.isCondition}>
+          <div
+            className={classnames(classes.innerContainer, {
+              [classes.split]: !selectedNode.isSource && !selectedNode.isSink,
+            })}
+          >
+            <h2 className={classes.h2Title}>{T.translate(`${I18N_PREFIX}.inputHeader`)}</h2>
+            {showInputTabs
+              ? getTabs(getTabConfig(inputs, selectedRecord))
```

Comment:
edge case to test would be how overflow will be handled for the tabs (if there are a lot of inputs our outputs)

Reply:
Good call out! I've added styling to make sure the tabs scroll if there is overflow.